roles/openshift_master/tasks/bootstrap.yml

```diff
@@ -26,3 +26,66 @@
   copy:
     content: "{{ kubeconfig_out.stdout }}"
     dest: "{{ openshift_master_config_dir }}/bootstrap.kubeconfig"
+
+- name: create a temp dir for this work
+  command: mktemp -d /tmp/openshift_node_config-XXXXXX
+  register: mktempout
+  run_once: true
+
+# This generate is so that we do not have to maintain
+# our own copy of the template.  This is generated by
+# the product and the following settings will be
+# generated by the master
+- name: generate a node-config dynamically
+  command: >
+    {{ openshift_master_client_binary }} adm create-node-config
+    --node-dir={{ mktempout.stdout }}/
+    --node=CONFIGMAP
+    --hostnames=test
+    --certificate-authority={{ openshift_master_config_dir }}/ca.crt
+    --signer-cert={{ openshift_master_config_dir }}/ca.crt
+    --signer-key={{ openshift_master_config_dir }}/ca.key
+    --signer-serial={{ openshift_master_config_dir }}/ca.serial.txt
+    --node-client-certificate-authority={{ openshift_master_config_dir }}/ca.crt
+  register: configgen
+  run_once: true
+
+- name: remove the default settings
+  yedit:
```

Comment:
The product needs to accept these overrides during config creation time.

Reply:
I'm not against that but I think it is too cumbersome for them to manage all of the options possible that a user could want.

Reply:
I don't want to create multiple configs.  Ansible should generate a one shot config *before* cluster configuration, then apply it to the cluster.  That's config creation time, and it can / should be done before the cluster is even booted.  I'd prefer that we create zero files on master or node at this point that don't already exist, since the end goal is to lay down nodes, boot a master, and then everything after that should be through the API.